scene/resources/mesh_data_tool.cpp

```diff
@@ -422,7 +430,11 @@ Vector<int> MeshDataTool::get_vertex_bones(int p_idx) const {
 
 void MeshDataTool::set_vertex_bones(int p_idx, const Vector<int> &p_bones) {
 	ERR_FAIL_INDEX(p_idx, vertices.size());
-	ERR_FAIL_COND(p_bones.size() != 4);
+	if (format & Mesh::ARRAY_FLAG_USE_8_BONE_WEIGHTS) {
+		ERR_FAIL_COND(p_bones.size() != 8);
+	} else {
+		ERR_FAIL_COND(p_bones.size() != 4);
+	}
```

Comment:
I'm not sure if `ERR_*` inside the `if` clause is a decent code style. I didn't see it used much elsewhere